module/os/linux/spl/Makefile.in

```diff
@@ -11,6 +11,7 @@ $(MODULE)-objs += ../os/linux/spl/spl-proc.o
 $(MODULE)-objs += ../os/linux/spl/spl-procfs-list.o
 $(MODULE)-objs += ../os/linux/spl/spl-taskq.o
 $(MODULE)-objs += ../os/linux/spl/spl-thread.o
+$(MODULE)-objs += ../os/linux/spl/spl-trace.o
```

Comment:
Yes, I agree.

I think another potential alternative might be to duplicate some of the common logic between the "zfs" and "spl" modules, for the "tracing infrastructure". I didn't spend too much time trying to figure out how best to do this, so let me know if you think I should look more into how to do this without violating the layers so much.

Reply:
It would be nice if we could pull only the "tracing infrastructure" in to the spl layer.  Then the spl layer could define the probes it needed, and the zfs layer could do likewise.  This might be possible with a little refactoring of the headers and I think it would be worth looking in to.

This would be nice since it would align the tracing functionality with how it's handled on the other platforms which provide dtrace as part of system.

Reply:
@behlendorf I took a stab at this, let me know what you think after you have time to give it a look.

I tried to keep the prior version available so you can see the differences more clearly (e.g. if you `git fetch` and `git diff` locally), but when we're ready to land, we can remove the first two commits in this PR; the original implementation and the revert.